Documentation/specs/installation-scenarios.md

```diff
@@ -0,0 +1,243 @@
+Each official, successful build should create and upload packages to location described by following URL:
+```
+https://dotnetcli.blob.core.windows.net/dotnet/<Channel>/<Version>/dotnet.<OSID>.<Version>.<Extension>
+```
+Content of the package should contain binaries which layout will be described later.
+
+Additionally each build should update `latest` [version descriptors](#version-descriptors)
+
+## Windows output
+
+Nuget - WIP, this should include versioning scheme
+
+| `<Extension>` | Description |
+| --- | :--- |
+| exe | Installer bundle. It should be used by end customers |
+| zip | Packed binaries. Used by [installation script](#installation-scripts) |
+| symbols.zip | Packed binaries with included symbols. See [symbol packages](#symbol-packages) |
+
+### Including dotnet cli installer as part of your bundle
+
+In order to install dotnet cli with other installer you need an MSI package. To get MSI, download exe bundle and extract it using `dark` tool which is part of [WiX Toolset](http://wixtoolset.org):
+```
+dark.exe -x <FolderWhereToExtract> <InstallerPath>
+```
+
+## OSX output
+
+| `<Extension>` | Description |
+| --- | :--- |
+| pkg | WIP |
+| tar.gz | Packed binaries. Used by [installation script](#installation-scripts) |
+| symbols.tar.gz | Packed binaries with included symbols. See [symbol packages](#symbol-packages) |
+
+## Ubuntu output
+
+| `<Extension>` | Description |
+| --- | :--- |
+| deb | Debian package. This package is being pushed to a [debian feed](#debian-feed) |
+| tar.gz | Packed binaries. Used by [installation script](#installation-scripts) |
+| symbols.tar.gz | Packed binaries with included symbols. See [symbol packages](#symbol-packages) |
+
+## RedHat/CentOS output
+
+| `<Extension>` | Description |
+| --- | :--- |
+| tar.gz | Packed binaries. Used by [installation script](#installation-scripts) |
+| symbols.tar.gz | Packed binaries with included symbols. See [symbol packages](#symbol-packages) |
+
+## Debian output
+
+| `<Extension>` | Description |
+| --- | :--- |
+| tar.gz | Packed binaries. Used by [installation script](#installation-scripts) |
+| symbols.tar.gz | Packed binaries with included symbols. See [symbol packages](#symbol-packages) |
+
+## Example build output links
+WIP
+
+## Questions
+- Should <Version> include channel name to avoid situation where you have two files on your computer and latest file might have lower version than the newest?
+
+# Obtaining dotnet
+
+## Installation scripts
+
+Installation script is a shell script which lets customers install dotnet.
+
+For Windows we are using PowerShell script (install-dotnet.ps1).
+For any other OS we are using bash script (install-dotnet.sh)
+
+WIP: Exact script action description.
+
+### Script arguments description
+
+| PowerShell script | Bash script | Default | Description |
+| --- | --- | --- | --- |
+| -Channel | --channel | production | Which [channel](#channels) to install from. Possible values: `future`, `preview`, `production` |

```

Comment:
We don't use powershell in corefx, but I've found that having different syntax for different scripting languages tends to be confusing, without any real benefit. I know some of the build scripts in here already use this style, but since these installer scripts would be more "public-facing" (i.e. anyone would use them, not just devs in the repo), we might want to consider making them uniform across platforms.